services/scanner/agentpool/poolagent/agent.go

```diff
@@ -219,46 +226,88 @@ func (agent *Agent) SetClient(agentClient clients.AgentClient) {
 // from request channels.
 func (agent *Agent) StartProcessing() {
 	agent.initWait.Add(1)
-	go agent.initialize()
 
+	go func() {
+		err := agent.startInitWorker(agent.ctx)
+		if err != nil {
+			log.WithError(err).WithField("bot-id", agent.config.ID).Warn("failed to initialize bot")
+		}
+	}()
 	go agent.processTransactions()
 	go agent.processBlocks()
 	go agent.processCombinationAlerts()
 }
 
-func (agent *Agent) initialize() {
-	defer agent.initWait.Done()
+func (agent *Agent) startInitWorker(ctx context.Context) error {
+	bExp := &backoff.ExponentialBackOff{
+		InitialInterval:     time.Second * 30,
+		RandomizationFactor: backoff.DefaultRandomizationFactor,
+		Multiplier:          backoff.DefaultMultiplier,
+		MaxInterval:         time.Minute * 10,
+		MaxElapsedTime:      time.Hour,
+		Stop:                backoff.Stop,
+		Clock:               backoff.SystemClock,
+	}
 
-	logger := log.WithFields(log.Fields{
-		"agent": agent.config.ID,
-	})
+	bo := backoff.WithContext(bExp, ctx)
+	err := backoff.Retry(
+		func() error {
+			initCtx, cancel := context.WithTimeout(agent.ctx, DefaultAgentInitializeTimeout)
+			defer cancel()
+			return agent.initialize(initCtx)
+		}, bo,
+	)
 
-	ctx, cancel := context.WithTimeout(agent.ctx, DefaultAgentInitializeTimeout)
-	defer cancel()
-	initializeResponse, err := agent.client.Initialize(ctx, &protocol.InitializeRequest{
-		AgentId:   agent.config.ID,
-		ProxyHost: config.DockerJSONRPCProxyContainerName,
-	})
+	return fmt.Errorf("backoff initialize failed: %v", err)
+}
+
+func (agent *Agent) initialize(ctx context.Context) error {
+	logger := log.WithFields(
+		log.Fields{
+			"agent": agent.config.ID,
+		},
+	)
+
+	initializeResponse, err := agent.client.Initialize(
+		ctx, &protocol.InitializeRequest{
+			AgentId:   agent.config.ID,
+			ProxyHost: config.DockerJSONRPCProxyContainerName,
+		},
+	)
 
 	if status.Code(err) == codes.Unimplemented {
 		logger.WithError(err).Info("initialize() method not implemented in bot - safe to ignore")
-		return
+		return err
```

Comment:
Should we return nil here? We successfully detected that the bot already doesn't implement the method so no need to retry.

Reply:
```suggestion
		return nil
```